pkg/controller/kfservice/resources/knative/service.go

```diff
@@ -171,3 +170,81 @@ func (c *ServiceBuilder) CreatePredictorService(name string, metadata metav1.Obj
 
 	return service, nil
 }
+
+func (c *ServiceBuilder) CreateTransformerService(name string, metadata metav1.ObjectMeta, transformerSpec *v1alpha2.TransformerSpec, isCanary bool) (*knservingv1alpha1.Service, error) {
+	annotations := utils.Filter(metadata.Annotations, func(key string) bool {
+		return !utils.Includes(serviceAnnotationDisallowedList, key)
+	})
+
+	if transformerSpec.MinReplicas != 0 {
+		annotations[autoscaling.MinScaleAnnotationKey] = fmt.Sprint(transformerSpec.MinReplicas)
+	}
+	if transformerSpec.MaxReplicas != 0 {
+		annotations[autoscaling.MaxScaleAnnotationKey] = fmt.Sprint(transformerSpec.MaxReplicas)
+	}
+
+	// User can pass down scaling target annotation to overwrite the target default 1
+	if _, ok := annotations[autoscaling.TargetAnnotationKey]; !ok {
+		annotations[autoscaling.TargetAnnotationKey] = constants.DefaultScalingTarget
+	}
+	// User can pass down scaling class annotation to overwrite the default scaling KPA
+	if _, ok := annotations[autoscaling.ClassAnnotationKey]; !ok {
+		annotations[autoscaling.ClassAnnotationKey] = autoscaling.KPA
+	}
+
+	predictorHostName := constants.DefaultPredictorServiceName(metadata.Name)
+	if isCanary {
+		predictorHostName = constants.CanaryPredictorServiceName(metadata.Name)
+	}
+	service := &knservingv1alpha1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: metadata.Namespace,
+			Labels:    metadata.Labels,
+		},
+		Spec: knservingv1alpha1.ServiceSpec{
+			ConfigurationSpec: knservingv1alpha1.ConfigurationSpec{
+				Template: &knservingv1alpha1.RevisionTemplateSpec{
+					ObjectMeta: metav1.ObjectMeta{
+						Labels: utils.Union(metadata.Labels, map[string]string{
+							constants.KFServicePodLabelKey: metadata.Name,
+						}),
+						Annotations: annotations,
+					},
+					Spec: knservingv1alpha1.RevisionSpec{
+						RevisionSpec: v1beta1.RevisionSpec{
+							// Defaulting here since this always shows a diff with nil vs 300s(knative default)
+							// we may need to expose this field in future
+							TimeoutSeconds: &constants.DefaultTimeout,
+							PodSpec: v1.PodSpec{
+								ServiceAccountName: transformerSpec.ServiceAccountName,
+								Containers: []v1.Container{
+									{
+										Image: transformerSpec.Custom.Container.Image,
+										Args: []string{
+											constants.ModelServerArgsModelName,
+											metadata.Name,
+											constants.ModelServerArgsPredictorHost,
+											predictorHostName,
```

Comment:
miss `protocol` args?

Reply:
currently it defaults to TensorFlow protocol, there is seldon protocol as well but it is not exposed on spec yet.